lib/approval_request_listener.rb

```diff
@@ -0,0 +1,67 @@
+require "manageiq-messaging"
+
+class ApprovalRequestListener
+  SERVICE_NAME = "platform.approval".freeze
+  CLIENT_AND_GROUP_REF = "approval-catalog-api-worker".freeze
+  EVENT_REQUEST_FINISHED = 'request_finished'.freeze
+
+  attr_accessor :messaging_client_options, :client
+
+  def initialize(messaging_client_options = {})
+    self.messaging_client_options = default_messaging_options.merge(messaging_client_options)
+  end
+
+  def run
+    Thread.new { subscribe_to_approval_updates }
+  end
+
+  def subscribe_to_approval_updates
+    self.client = ManageIQ::Messaging::Client.open(messaging_client_options)
+
+    client.subscribe_topic(
+      :service   => SERVICE_NAME,
+      :max_bytes => 500_000
+    ) do |messages|
+      messages.each do |msg|
+        process_message(msg)
+      end
+    end
+  ensure
+    client&.close
+    self.client = nil
+  end
+
+  private
+
+  def process_message(msg)
+    ProgressMessage.create!(
```

Comment:
@syncrou Should this be calling the update_message in https://github.com/ManageIQ/catalog-api/blob/0db1da44e21b1fef1bfa7934d6bd4679d9905950/app/models/order_item.rb#L42

Reply:
@syncrou This needs to be tied back to an order item. From the approval request you have to pickup the order_item and call progress message on the order item.
If the order item or request is not found we should log that as an exception, if we don't find the record I think it would mean we picked up someone else's message. If we have 2 pods running but using different databases. Typically when a dev user is running a pod with his own database and uses the same worker names.